metaseq/modules/multihead_attention.py

```diff
@@ -312,6 +314,24 @@ def forward(
             # In this branch incremental_state is never None
             assert incremental_state is not None
             incremental_state = self._set_input_buffer(incremental_state, saved_state)
+
+        if isinstance(q, ShardedTensor):
```

Comment:
Let's add comments here.

Reply:
tgt_len, bsz, embed_dim = query.size()

after linear

tgt_len, bsz, embed_dim // paralle_size - col-wise sharding original q size. sharded by dim 2

tgt_len, bsz * self.num_heads_partition, self.head_dim implicitly change the sharding dim to 1.